packages/js/data/src/products/selectors.ts

```diff
@@ -122,38 +121,24 @@ export const isPending = (
 	return false;
 };
 
-export const getPermalinkParts = createRegistrySelector(
-	( select ) => ( state: ProductState, productId: number ) => {
-		const product = select( 'core' ).getEntityRecord(
-			'postType',
-			'product',
-			productId,
-			// @ts-expect-error query object is not part of the @wordpress/core-data types yet.
-			{
-				_fields: [
-					'id',
-					'permalink_template',
-					'slug',
-					'generated_slug',
-				],
-			}
-		);
-		if ( product && product.permalink_template ) {
-			const postName = product.slug || product.generated_slug;
+export const getPermalinkParts = ( state: ProductState, productId: number ) => {
+	const product = state.data[ productId ];
 
-			const [ prefix, suffix ] = product.permalink_template.split(
-				PERMALINK_PRODUCT_REGEX
-			);
+	if ( product && product.permalink_template ) {
+		const postName = product.slug || product.generated_slug;
 
-			return {
-				prefix,
-				postName,
-				suffix,
-			};
-		}
-		return null;
+		const [ prefix, suffix ] = product.permalink_template.split(
+			PERMALINK_PRODUCT_REGEX
+		);
+
+		return {
+			prefix,
+			postName,
+			suffix,
+		};
```

Comment:
Given this returns a new object which isn't easily memoizable, it may be worth wrapping this in a `createSelector` function with the dependency being `state.data[ productId ]`, see the `getProducts` selector above for an example.

Reply:
Thanks, @louwie17 -- I'll merge this PR and create a small follow-up to take care of that.